app/services/dqt/register_for_trn.rb

```diff
@@ -33,7 +33,7 @@ def submit_trn_request
     end
 
     def update_trainee
-      trainee.submit_for_trn!
+      trainee.submit_for_trn! if trainee.can_submit_for_trn?
```

Comment:
I'm a bit confused as to why we call `trainee.submit_for_trn!` in `Trainees::SubmitForTrn` and another call here in `Dqt::RegisterForTrn`.

I guess `trainee.can_submit_for_trn?` will always return false at this point because the state change has already happened?

We need to decide at which point the trainee should be transitioned. I think that we originally had it one level up in `Trainees::SubmitForTrn` because if the request did fail for some reason, we still wanted the user to see the trainee in the correct state. We would separately be alerted to fix the actual issue via Sentry. cc @gpeng - am I remembering this correctly?

Reply:
Actually, I guess it's ok to have them both here in case we want to call this service separately, maybe I've unconfused myself.

Reply:
Coolio, it just seemed a bit strange to have it being called twice. Agreed that it doesn't do any harm it being here though 👍  happy to keep it as it is.